index.js

```diff
@@ -0,0 +1,61 @@
+const { exec } = require('child_process');
+const fs = require('fs');
+
+const refreshRate = 100;
+const logFilePath = 'activityMonitor.log';
+
+function executeCommand(command) {
+	return new Promise((resolve, reject) => {
+		exec(command, (error, stdout, stderr) => {
+			if (error) {
+				reject(error);
+				return;
+			}
+			resolve(stdout.trim());
+		});
+	});
+}
+
+function writeToLog(data) {
+	const currentTime = new Date().getTime();
+	const logEntry = `${currentTime} : ${data}\n`;
+
+	fs.appendFile(logFilePath, logEntry, (err) => {
+		if (err) {
+			console.error(`Error: ${err}`);
+		}
+	});
+}
+
+async function monitor() {
+	let prevLineData = null;
+
+	setInterval(async () => {
+		try {
+			const platform = process.platform;
+			let command = '';
+
+			if (platform === 'win32') {
```

Comment:
Not critical, but: 
it's better to use configuration objects for such switchers to not write conditions inside the main function implementation. Like:
``` js
// config could be moved to separate file.
const config = {command: isWindows ? '...' : '...'} 
....
log(config.command)
```

Reply:
I created 2 constants